### DISCUSSION TITLE
20241212 Weekly Development Meeting

### DISCUSSION THREAD

**Discussion by josecelano** [nautilus-cyberneering/.github#126] [Meetings]
- Time:  9:30 in Canary Islands (WET (UTC+0)).
- Time: 10:30 in Vaduz (CET (UTC+1)).

Location:
<https://meeting.blockfinance-eco.li/NautilusCyberneeringWeeklyDevMeeting>

Expected duration: 2 to 4 hours with breaks

Communicated attendants:

- Cameron Garnham(@da2ce7)  - x 
- Jose Celano(@josecelano) - x
- Constantin Bosse(@cgbosse) - x
- Mario Melchor(@mario-nt) - x
- Graeme Byrne(@grmbyrn) - x

-------------------------------------------------------------

## Agenda:

1. Everyone will take their turn and:
- Go over his notes added as comments for the meeting which will have been added in advance to give an update of all the work done during the previous week indicating.
	- The issues worked on
	- The progress made on them
	- Other things done
- Inform of any faced challenges and request help if necessary to work on specific issues.
- Highlight the encountered difficulties, especially if there is a risk of them causing delays or affecting end product's quality.
- Communicate what will be the next week's tasks.
- List any new topics for further discussion in an additional Technical meeting that he noted for future discussion in comments to this discussion either before the meeting or after his turn so that they get considered.

During the meeting every team member will take their turns and follow this sequence.
The topics for further discussion will be noted in advance by each the team member and to the meetings agenda.

2. At the end of the meeting if there any new technical topics to be discussed further it will be decided to
set another meeting date for the discussion of these or to hold the meeting hereafter.

Also the persons to be part of the meeting will be decided on so that the others will be free to continue their own work.

**Comment by josecelano** [comment]
### Torrust Tracker

Issues:

- [ ] https://github.com/torrust/torrust-tracker/issues/1096 (with a Counting Bloom Filter)
- [ ] https://github.com/torrust/torrust-tracker/issues/1126 (needed to setup Prometheus and Grafana)

[PR](https://github.com/torrust/torrust-tracker/pull/1124) for the first iteration is pending to review by @da2ce7.

### Torrust Index

Comment on issues:

- https://github.com/torrust/torrust-index/issues/778#issuecomment-2520691235 (and comments after this)

@da2ce7 I'm waiting for your feedback to continue working on this. Maybe we can discuss it in the meeting.

### Torrust Demo

Issues:

- [ ] https://github.com/torrust/torrust-demo/issues/20

I did some maintenance before adding the new services:

- I've updated the [repo](https://github.com/torrust/torrust-demo) with the docker-compose configuration we are using in production.
- I've updated the server to use this repo.
- I've added two new services to the compose configuration: Prometheus and Grafana
- I've deployed these new two services. Prometheus is not public, and Grafana is on http://grafana.torrust-demo.com:3100/

**TODO**:

- Set up HTTPs for Grafana using the Nginx proxy ([more info](https://github.com/torrust/torrust-demo/issues/20)).

The Grafana dashboard after collecting data for 2 hours:

![image](https://github.com/user-attachments/assets/387f255f-ac7d-48a3-8a00-a4a3ea947ce3)

@da2ce7 I emailed you the Grafana credentials.

**Note:** time spent on Torrust (26h)

**Comment by mario-nt** [comment]
### GUI

- https://github.com/torrust/torrust-index-gui/issues/658

### Tracker
- https://github.com/torrust/torrust-tracker/issues/1069

**Comment by grmbyrn** [comment]
## Torrust Website

* Migrate from Svelte 4 to Svelte 5 as mentioned in this https://github.com/torrust/torrust-website/issues/106.
* Update dependencies in package.json such as Vite which were preventing the successful migration to Svelte 5.
* Upgrade ESLint to v9 as v8 has recently reached end of life and is no longer maintained, as mentioned [here](https://eslint.org/docs/latest/use/migrate-to-9.0.0).
* Upgrade Sass as current syntax is deprecated as mentioned [here](https://sass-lang.com/documentation/breaking-changes/legacy-js-api/)
* Remove /contributors route as it is redundant since we now display GitHub contributors on home page and on Community page. Existing individual contributor pages have been retained and are accessible by clicking on the author's name in the blog post.
* Fix problem of when navigating to a new page, the site wouldn't scroll up to the top as would be expected. As found https://github.com/sveltejs/kit/pull/8723, this seems to be a Svelte issue.
* Remove Histoire as it currently isn't compatible with Vite 6.
